File: 2021/csharp/Aoc2021.Tests/Day03Tests.cs
namespace Aoc2021.Tests;

public class Day03Tests
{
    private static readonly List<string> report = new()
    {
        "00100",
        "11110",
        "10110",
        "10111",
        "10101",
        "01111",
        "00111",
        "11100",
        "10000",
        "11001",
        "00010",
        "01010",
    };

    [Fact]
    public void Rating1Works()
    {
        Assert.Equal(new(22, 9), Day03.PowerConsumtionRate.FromReport(report));
    }

    [Fact]
    public void Rating2Works()
    {
        Assert.Equal(new(23, 10), Day03.LifeSupportRate.FromReport(report));
    }

    [Fact]
    public void Stars()
    {
        Day03 run = new();
        Assert.Equal("1071734", run.Result1());
        Assert.Equal("6124992", run.Result2());
    }
}

File: 2021/fsharp/Aoc2021.Tests/Day03.Tests.fs
module Day03.Tests

open Xunit

let report =
    [ "00100"
      "11110"
      "10110"
      "10111"
      "10101"
      "01111"
      "00111"
      "11100"
      "10000"
      "11001"
      "00010"
      "01010" ]

[<Fact>]
let ``Day03 power consumption works`` () =
    Assert.Equal({ Gamma = 22; Epsilon = 9 }, report |> PowerConsumtionRate.ofReport)

[<Fact>]
let ``Day03 life support works`` () =
    Assert.Equal(
        { OxygenGenerator = 23
          CO2Scrubber = 10 },
        report |> LifeSupportRate.ofReport
    )

[<Fact>]
let ``Day03 Stars`` () =
    Assert.Equal("1071734", job1 ())
    Assert.Equal("6124992", job2 ())
